src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -2641,6 +2640,20 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
       val selectorTp = packCaptured(selector1.tpe.widen).skolemizeExistential(context.owner, selector)
       val casesTyped = typedCases(cases, selectorTp, pt)
 
+      def initChildren(sym: Symbol): Unit = if (sym.isJava && sym.isSealed) {
+        sym.attachments.get[PermittedSubclassSymbols].foreach(_.permits.foreach { child =>
+          child.initialize
+          initChildren(child)
+        })
+        patmat.javaClassesByUnit.get(sym.pos.source).foreach(_.foreach { c =>
+          if (c.parentSymbols.contains(sym)) {
+            c.initialize
+            initChildren(c)
```

Comment:
love that `c` is both `class` and `child`, brilliant naming

Reply:
also hoping lrytz doesn't mind that I updated the brackets. As a vi user, the advantage of having the LBRACE trailing on the line is for $% to match them. What a hassle to f{ first. This mandates small blocks, which ought to be a rule anyway.